termd-build-driver/src/main/java/org/jboss/pnc/termdbuilddriver/commands/TermdCommandInvoker.java

```diff
@@ -0,0 +1,103 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.termdbuilddriver.commands;
+
+import org.jboss.pnc.termdbuilddriver.statusupdates.TermdStatusUpdatesConnection;
+import org.jboss.pnc.termdbuilddriver.statusupdates.event.Status;
+import org.jboss.pnc.termdbuilddriver.statusupdates.event.UpdateEvent;
+import org.jboss.pnc.termdbuilddriver.websockets.TermdTerminalConnection;
+
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.nio.file.Path;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.LinkedBlockingQueue;
+
+public class TermdCommandInvoker {
+
+    private TermdTerminalConnection termdTerminalConnection;
+    private TermdStatusUpdatesConnection termdStatusUpdatesConnection;
+    private URI baseServerUri;
+    private Path workingDirectory;
+
+    private BlockingQueue<UpdateEvent> eventQueue = new LinkedBlockingQueue<>();
+    private Queue<InvocatedCommandResult> invokedCommands = new LinkedList<>();
+
+    public TermdCommandInvoker(URI serverBaseUri, Path workingDirectory) {
+        this.baseServerUri = serverBaseUri;
+        this.termdTerminalConnection = new TermdTerminalConnection(serverBaseUri);
+        this.termdStatusUpdatesConnection = new TermdStatusUpdatesConnection(serverBaseUri);
+        this.workingDirectory = workingDirectory;
+    }
+
+    /**
+     * Performs command and the invocation is successful - returns proper result.
+     *
+     * Otherwise a {@link org.jboss.pnc.termdbuilddriver.TermdException} or its subclasses will be thrown and should be handled
+     * using {@link java.util.concurrent.CompletableFuture#exceptionally(java.util.function.Function)}.
+     *
+     * @param command Command to be executed.
+     * @return Completable future representing the result.
+     */
+    public CompletableFuture<InvocatedCommandResult> performCommand(String command) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                String data = "{\"action\":\"read\",\"data\":\"" + command + "\\r\\n\"}";
+                termdTerminalConnection.sendAsBinary(ByteBuffer.wrap(data.getBytes()));
+                InvocatedCommandResult invocatedCommandResult = new InvocatedCommandResult(eventQueue.take(), baseServerUri, getLogsDirectory());
+                invokedCommands.add(invocatedCommandResult);
+                return invocatedCommandResult;
+            } catch (Exception e) {
+                throw new TermdCommandExecutionException("There was a problem when invoking command " + command, e);
+            }
+        });
+    }
+
+    protected String getLogsDirectory() {
+        return workingDirectory.toAbsolutePath().toString();
```

Comment:
Log directory is not linked to working directory it is defined via agent boot command line. We should define it in setting on driver side too.